examples/src/main/scala/org/apache/spark/examples/mllib/LDAExample.scala

```diff
@@ -128,7 +133,16 @@ object LDAExample {
 
     // Run LDA.
     val lda = new LDA()
-    lda.setK(params.k)
+
+    val optimizer = params.algorithm match {
+      case "em" => new EMLDAOptimizer
+      case "online" => new OnlineLDAOptimizer().setMiniBatchFraction(0.05 + 1.0 / actualCorpusSize)
```

Comment:
Please add a small comment saying the adjustment is to be more robust on tiny datasets.